pkg/operator/client/deploy.go

```diff
@@ -481,6 +491,60 @@ func getRemovedCharts(prevDir string, curDir string, previousKotsCharts []v1beta
 	return removedCharts, nil
 }
 
+func findMatchingV1Beta1Charts(chartsDir string, kotsCharts []kotsutil.HelmChartInterface) ([]kotsutil.HelmChartInterface, error) {
+	dirContent, err := ioutil.ReadDir(chartsDir)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to list chart dir %s", chartsDir)
+	}
+
+	matching := []kotsutil.HelmChartInterface{}
+
+	for _, f := range dirContent {
+		if !f.IsDir() {
+			continue
+		}
+
+		dirName := f.Name()
+		chartDir := filepath.Join(chartsDir, dirName)
+		chartName, chartVersion, err := findChartNameAndVersion(chartDir)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to find chart name and version in %s", chartDir)
+		}
+
+		for _, kotsChart := range kotsCharts {
+			if kotsChart.GetChartVersion() == chartVersion && kotsChart.GetChartName() == chartName && kotsChart.GetDirName() == dirName {
+				matching = append(matching, kotsChart)
+			}
+		}
+	}
+
+	return matching, nil
+}
+
+func findMatchingV1Beta2Charts(chartsDir string, kotsCharts []kotsutil.HelmChartInterface) ([]kotsutil.HelmChartInterface, error) {
```

Comment:
why does this need to be a different function? why not just rename and use the `findMatchingV1Beta1Charts` function?

Reply:
nice catch.  looks like this was from before the interface was created.  removing...

Reply:
it's different since v1beta1 charts are in a directory, but the v1beta2 chart is just the tgz archive plus a user-provided values.yaml file

Reply:
changed it to be a common function that will act based on the `apiVersion`